AbilityDeck.jsx

```diff
@@ -86,22 +130,143 @@ export default function AbilityDeck(props) {
+          </tr>
+        )
+        : (
+          <tr>
+            <td className={css.baseStatLeft}>{render_val(data[0])}</td>
+            <td className={css.baseStatId}>
+              <StatIcon name={iconName} className={css.intStatIcon} />
+            </td>
+            <td className={css.baseStatRight}>{render_val(data[1])}</td>
+          </tr>
+        );
+    };
+
+    const boolStatLine = (mobType, data) => {
+      if (!data.length) {
+        return null;
+      }
+
+      const icons = data.map(iconName => <StatIcon name={iconName} className={css.boolStatIcon} />);
+      return (
+        <tr className={css.boolStatLine}>
+          <td>{mobType}</td>
+          <td colSpan="3">{icons}</td>
+        </tr>
+      );
+    };
+
+    const attr_lines = Object.keys(int_attr_dict).map(key =>
+      intStatLine(key, int_attr_dict[key]));
+
+    return (
+      <div className={css.baseStats}>
+        <table>
+          { intStatLine('heal', wrap_array(health)) }
+          { intStatLine('move', wrap_array(move)) }
+          { intStatLine('attack', wrap_array(attack)) }
+          { intStatLine('range', wrap_array(range)) }
+          {attr_lines}
+          { boolStatLine('N', bool_attrs[0]) }
+          { boolStatLine('E', bool_attrs[1]) }
+        </table>
+      </div>
+    );
+  };
+
+  const renderBossExtras = () => {
+    const text_attrs = Object.keys(txt_attr_dict).map((key) => {
+      const who = txt_attr_dict[key];
+      let modifier = null;
+      if (!who[0]) {
+        modifier = ' (elite only)';
+      } else
+      if (!who[1]) {
+        modifier = ' (normal only)';
+      }
+      return <li>{key}{modifier}</li>;
+    });
+
+    const immunity_line = () => {
+      if (!stats.immunities) {
+        return null;
+      }
+
+      const Icon = x => <StatIcon name={x} className={css.boolStatIcon} />;
+      const icons = stats.immunities
+        ? Object.keys(stats.immunities).map(key => Icon(stats.immunities[key]))
+        : null;
+      return <li>Immunities: {icons}</li>;
+    };
+
+    const notes = () => (stats.notes ? <li>{stats.notes}</li> : null);
+
+    return (
+      <div className={css.baseStats}>
+        <ul>
+          { text_attrs }
+          { immunity_line() }
+          { notes() }
+        </ul>
+      </div>
+    );
+  };
+
   const [topDraw] = props.deckState.draw_pile;
   const [topDiscard, sndDiscard] = props.deckState.discard;
   return (
-    <ButtonDiv
-      className={props.hidden ? css.hiddenDeck : css.cardContainer}
-      onClick={props.onClick}
+    <div
+      className={props.hidden ? css.hiddenDeck : null}
     >
-      {topDraw ? renderCard(topDraw, -7, [css.draw], false) : null}
-      {topDiscard ? renderCard(topDiscard, -3, [css.discard, css.pull], true) : null}
-      {sndDiscard ? renderCard(sndDiscard, -4, [css.discard, css.lift], true) : null}
-    </ButtonDiv>
+      <div className={css.rowContainer}>
```

Comment:
This breaks the unfortunately fragile `refresh_ui()` function in Tableau.jsx, which is what handles card text scaling for different size windows.  I'm not particularly opposed to the structure changes here, but we should update `refresh_ui()` to deal with it.  I'll open a separate issue to track the improvement of getting rid of `refresh_ui`.

The quick test for the resizing logic is to flip a card over and resize your window until the text and cards start shrinking, which should happen when the window is too narrow for a single ability deck

Reply:
Oh, also I'd recommend rebasing before fixing that, since I just landed a fix for a different issue in refresh_ui()

Reply:
k, I think that I tracked it down.  The problem wasn't really with refresh_ui, it was the fact that the container holding the card was not shrinking when the window became too narrow.  If you slap a max-width on it which says that it cannot be wider than the view port, the card can shrink again, the the font scaling hack seems to do its job again.